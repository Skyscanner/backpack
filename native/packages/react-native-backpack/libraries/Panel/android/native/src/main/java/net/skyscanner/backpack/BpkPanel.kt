package net.skyscanner.backpack

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout


class BpkPanel(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int) : LinearLayout(context, attrs, defStyleAttr) {

    constructor(context: Context) : this(context, null)
    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, R.style.Bpk_panel)

    init {
        this.orientation = LinearLayout.VERTICAL
        initialize(context, attrs, defStyleAttr)
    }

    var mPadding: Boolean = false
        set(value) {
            field = value
            draw()
        }

    private fun draw() {
        this.background = context.getDrawable(R.drawable.border)
        val padding = resources.getDimensionPixelOffset(R.dimen.bpkSpacingSm)
        if (mPadding) {
            this.setPadding(padding, padding, padding, padding)
        }
        val params = LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT)
        params.setMargins(padding, padding, padding, padding)
        this.layoutParams = params
    }


    private fun initialize(context: Context, attrs: AttributeSet?, defStyleAttr: Int) {

        val a = context.obtainStyledAttributes(attrs, R.styleable.panel, R.attr.bpk_padding, defStyleAttr)
        if (a.hasValue(R.styleable.panel_bpk_padding)) {
            //default value of padding is true
            mPadding = a.getBoolean(R.styleable.panel_bpk_padding, true)
        }
        draw()
    }

}
