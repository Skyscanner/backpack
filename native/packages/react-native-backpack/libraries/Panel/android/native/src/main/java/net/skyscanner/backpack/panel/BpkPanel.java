package net.skyscanner.backpack.panel;

import android.content.Context;
import android.view.View;

public class BpkPanel extends View {

  public BpkPanel(Context context) {
    super(context);
  }

  //TODO implementation
}
