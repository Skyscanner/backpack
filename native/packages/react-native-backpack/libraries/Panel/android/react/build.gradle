apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'

android {
  compileSdkVersion rootProject.ext.compileSdkVersion
  buildToolsVersion rootProject.ext.buildToolsVersion


  defaultConfig {
    minSdkVersion rootProject.ext.minSdkVersion
    targetSdkVersion rootProject.ext.targetSdkVersion
    versionCode 1
    versionName "1.0"
  }
}

dependencies{
  implementation "com.facebook.react:react-native:+" // From node_modules
  implementation project(':panel-native')
  implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
}
repositories {
  mavenCentral()
}
