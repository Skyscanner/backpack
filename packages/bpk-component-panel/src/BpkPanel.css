@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-panel{display:block;background-color:#fff;box-shadow:0 0 0 1px #dddde5 inset;border-radius:.75rem}.bpk-panel--padded{padding:.75rem}.bpk-panel--full-width{border-radius:0;box-shadow:#dddde5 0 -.0625rem 0 0 inset,#dddde5 0 .0625rem 0 0 inset}
