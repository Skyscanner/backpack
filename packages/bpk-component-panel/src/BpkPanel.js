/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';
import React from 'react';

import { cssModules } from '../../bpk-react-utils';

import STYLES from './BpkPanel.module.scss';

const getClassName = cssModules(STYLES);

const BpkPanel = (props) => {
  const classNames = [getClassName('bpk-panel')];
  const { children, className, fullWidth, padded, ...rest } = props;

  if (padded) {
    classNames.push(getClassName('bpk-panel--padded'));
  }
  if (fullWidth) {
    classNames.push(getClassName('bpk-panel--full-width'));
  }
  if (className) {
    classNames.push(className);
  }

  return (
    <section className={classNames.join(' ')} {...rest}>
      {children}
    </section>
  );
};

BpkPanel.propTypes = {
  children: PropTypes.node.isRequired,
  padded: PropTypes.bool,
  fullWidth: PropTypes.bool,
  className: PropTypes.string,
};

BpkPanel.defaultProps = {
  padded: true,
  fullWidth: false,
  className: null,
};

export default BpkPanel;
