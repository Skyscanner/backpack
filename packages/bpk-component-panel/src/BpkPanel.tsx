/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import type { ReactNode } from 'react';

import { cssModules } from '../../bpk-react-utils';

import STYLES from './BpkPanel.module.scss';


const getClassName = cssModules(STYLES);

export type Props = {
  children: ReactNode;
  padded?: boolean,
  fullWidth?: boolean,
  className?: string | null,
  keyline?: boolean,
  [rest: string]: any;
};


const BpkPanel = ({
  children,
  className = null,
  fullWidth = false,
  keyline = true,
  padded = true,
  ...rest
}: Props) => {
  const classNames = [getClassName('bpk-panel')];

  if (padded) {
    classNames.push(getClassName('bpk-panel--padded'));
  }
  if (fullWidth) {
    classNames.push(getClassName('bpk-panel--full-width'));
  }
  if (keyline) {
    if (fullWidth) {
      classNames.push(getClassName('bpk-panel--full-width-keyline'));
    } else {
      classNames.push(getClassName('bpk-panel--keyline'));
    }
  }
  if (className) {
    classNames.push(className);
  }

  return (
    <section className={classNames.join(' ')} {...rest}>
      {children}
    </section>
  );
};

export default BpkPanel;
