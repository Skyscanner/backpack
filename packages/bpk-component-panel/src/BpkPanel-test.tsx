/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render } from '@testing-library/react';

import BpkPanel from './BpkPanel';

describe('BpkPanel', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <BpkPanel>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo
        ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis
        dis parturient montes, nascetur ridiculus mus.
      </BpkPanel>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with padded attribute', () => {
    const { asFragment } = render(
      <BpkPanel padded={false}>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo
        ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis
        dis parturient montes, nascetur ridiculus mus.
      </BpkPanel>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with fullWidth attribute', () => {
    const { asFragment } = render(
      <BpkPanel fullWidth>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo
        ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis
        dis parturient montes, nascetur ridiculus mus.
      </BpkPanel>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with className attribute', () => {
    const { asFragment } = render(
      <BpkPanel className="custom-class">
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo
        ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis
        dis parturient montes, nascetur ridiculus mus.
      </BpkPanel>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
