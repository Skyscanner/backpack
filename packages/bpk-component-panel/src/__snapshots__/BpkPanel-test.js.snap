// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkPanel should render correctly 1`] = `
<DocumentFragment>
  <section
    class="bpk-panel bpk-panel--padded bpk-panel--keyline"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
  </section>
</DocumentFragment>
`;

exports[`BpkPanel should render with className attribute 1`] = `
<DocumentFragment>
  <section
    class="bpk-panel bpk-panel--padded bpk-panel--keyline custom-class"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
  </section>
</DocumentFragment>
`;

exports[`BpkPanel should render with fullWidth attribute 1`] = `
<DocumentFragment>
  <section
    class="bpk-panel bpk-panel--padded bpk-panel--full-width bpk-panel--full-width-keyline"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
  </section>
</DocumentFragment>
`;

exports[`BpkPanel should render with no keyline when attribute is false 1`] = `
<DocumentFragment>
  <section
    class="bpk-panel bpk-panel--padded"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
  </section>
</DocumentFragment>
`;

exports[`BpkPanel should render with no keyline when attribute is false and fullwidth 1`] = `
<DocumentFragment>
  <section
    class="bpk-panel bpk-panel--padded bpk-panel--full-width"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
  </section>
</DocumentFragment>
`;

exports[`BpkPanel should render with padded attribute 1`] = `
<DocumentFragment>
  <section
    class="bpk-panel bpk-panel--keyline"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
  </section>
</DocumentFragment>
`;
